### ISSUE TITLE
[Bug]: Can't edit Docker Compose Location on a Docker Compose Build Pack project

### ISSUE THREAD

**Issue by DiegoBM** [coollabsio/coolify#2521]
### Description

When trying to add a project from a github repository using the Docker Compose build pack, it won't allow to change the docker compose location field which I need for it to find the file within the repo

### Minimal Reproduction (if possible, example repository)

* Setup a brand new server (in my case ubuntu server 24.04 LTS)
* Install docker
* Get Coolify up and running following the instructions in the website
* Setup coolify to install in localhost (no other settings)
* Create a new project
* Add a new resource, of type "Public Repository" and assign it to a server and destination
* Enter the repository URL (in my case https://github.com/supabase/supabase)
* Select "Docker Compose" as Build Pack and click continue
* On the next screen nothing is editable including "Base Directory" or "Docker Compose location" which I need to change for it to find the docker compose file in the repo. It also displays a "504 Gateway Time-out" after a few seconds

### Exception or Error

Build data should be editable, in particular Docker Compose Location

### Version

v4.0.0-beta.294

**Comment by DiegoBM** [comment]
Maybe you could add a depth setting to github repos, so that they can be cloned with depth 1?

**Comment by andrasbacsai** [comment]
Closing old issue (which is probably fixed already). Reopen if still reproducible.

**Comment by muuvmuuv** [comment]
Yes I still experience it, I am trying to configure it to `docker-compose.stage.yml` but it keeps reseting to name without "stage" in it. Therefor reloading fails. Our Stage file just extends the base file and removes/adds variables.
Edit: our version is v4.0.0-beta.454

**Comment by jevido** [comment]
I'm experiencing it as well on version:
v4.0.0-beta.454

**Comment by djedu28** [comment]
I'm experiencing it as well on version:
v4.0.0-beta.455

**Comment by djedu28** [comment]
@andrasbacsai The error still persists. 

You can check if it still happens in the current stable version: v4.0.0-beta.460. 

If it still happens, it is necessary to reopen the issue.

I'm on version 'v4.0.0-beta.455' and I didn't find any information related to fixing the bug in the update log '+v4.0.0-beta.455 -> v4.0.0-beta.460'

**Comment by djedu28** [comment]
@andrasbacsai   To replicate the BUG.

- Create a new project
- connect to a github repository that has Docker Compose files
- set 'Build Pack *' to 'Docker Compose'
    <img width="241" height="95" alt="Image" src="https://github.com/user-attachments/assets/36848a72-048d-4216-83f3-55e785fb5c34" />
- change 'Docker Compose Location' to a name other than '/docker-compose.yaml' (the file needs to exist)
    <img width="315" height="115" alt="Image" src="https://github.com/user-attachments/assets/e88d5ab3-b5ab-4dee-a98f-8fe17f66abf1" />

- Save the project, and receive pop-up successfully
    <img width="400" height="92" alt="Image" src="https://github.com/user-attachments/assets/e4fcfa11-29f2-46c5-8a07-d73173a9e09f" />

- Refresh the page. Verify that 'Docker Compose Location' has returned to default: '/docker-compose.yaml'
<img width="1221" height="195" alt="Image" src="https://github.com/user-attachments/assets/9a19df5c-311f-4e9f-b815-bf4855ddf65b" />

----

I don't know how. But at some point, the edit to '/docker-compose.config.yaml' was successfully performed, I generated the build and success. 

However, I can't edit the 'Docker Compose Location' again, for example, to '/docker-compose.yaml'.

https://github.com/user-attachments/assets/445e7d1b-edb7-43b7-b255-c35891fd94aa

PS.:
> (Before this video, and after successfully build, I deleted the file 'docker-compose.config.yaml' on github, hoping to force coolify to save the new address 'docker-compose.prod.yaml')

>When editing `Docker Compose Location`, and pressing "enter" (or clicking save), it's possible to preview the file's contents. The file's contents ('docker-compose.yaml') persist in the preview when refreshing the page, although the file name 'docker-compose.config.yaml' is displayed.

**Comment by Cinzya** [comment]
@djedu28 can you provide a link to an example repo that causes this bug for you. Also is this still happening on beta462 as well?

**Comment by HT-Moh** [comment]
This issue is really frustrating and easily reproducible, the docker location is always revert back to `yaml` after updating it to `yml` which cause the deploy to fail. this is a blocker. I won't use coolify at least for now.

**Comment by muuvmuuv** [comment]
I have installed a pgadmin as a tool and connect directly to the coolify db to manually override that entry ^^ that works fine for now.

**Comment by andrasbacsai** [comment]
I am trying to reproduce this issue, but I couldn't. 

Which version are you doing it @HT-Moh or @muuvmuuv ?

**Comment by HT-Moh** [comment]
I just tested again on v4.0.0-beta.459. The same issue. The extension and file name are not persistent.

Steps to reproduce:

Add a GitHub repo that contains docker-compose.yml instead of docker-compose.yaml.

Update Docker Compose Location to docker-compose.yml.

The Docker Compose content appears.

Refresh the page.

The Docker Compose Location resets to docker-compose.yaml.

Click deploy.

Error:
Deployment failed: Docker compose file not found at /docker-compose.yaml.
Check if the correct extension was used.

**Comment by Cinzya** [comment]
@DiegoBM @muuvmuuv @jevido @HT-Moh Can you check whether or not you have `SPA Navigation` enabled in the Coolify Settings and try this again after disabling it? This could be related to #9732 .

<img width="1029" height="936" alt="Image" src="https://github.com/user-attachments/assets/7653f957-7986-4656-9a7a-f4560c7b74df" />

**Comment by HT-Moh** [comment]
We still have the same issue.

The Docker name cannot be changed. The only workaround is to rename the Docker Compose file in my repo to match the default file name used by Coolify.

The second issue is more serious. I cannot add custom labels for Authentik setup. This blocks us from using Coolify for this use case.

There seems to be no way to add custom labels. I cannot edit the raw generated Docker config, and I do not see any menu or setting for this.

To be honest, we will decide soon if we keep using Coolify. For our use case, it may be better to keep full control and use Docker Swarm directly with Traefik, because some basic things we need are not working.

Current version: v4.0.0-beta.459
